plugins/linux-capture/drmsend.c

```diff
@@ -0,0 +1,187 @@
+#include "drmsend.h"
+
+#include <xf86drm.h>
+#include <libdrm/drm_fourcc.h>
+#include <xf86drmMode.h>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+
+#define ERR(fmt, ...) fprintf(stderr, "" fmt "\n", ##__VA_ARGS__)
+
+#define MSG(fmt, ...) fprintf(stdout, "obs-drmsend: " fmt "\n", ##__VA_ARGS__)
+
+void printUsage(const char *name)
+{
+	MSG("usage: %s /dev/dri/card socket_filename", name);
+}
+
+int main(int argc, const char *argv[])
+{
+	if (argc < 3) {
+		printUsage(argv[0]);
+		return 1;
+	}
+
+	const char *card = argv[1];
+	const char *sockname = argv[2];
+
+	MSG("Opening card %s", card);
+	const int drmfd = open(card, O_RDONLY);
+	if (drmfd < 0) {
+		perror("Cannot open card");
+		return 1;
+	}
+
+	drmSetClientCap(drmfd, DRM_CLIENT_CAP_UNIVERSAL_PLANES, 1);
+
+	int sockfd = -1;
+	int retval = 2;
+	drmsend_response_t response = {0};
+	int fb_fds[OBS_DRMSEND_MAX_FRAMEBUFFERS] = {-1};
+
+	{
+		drmModePlaneResPtr planes = drmModeGetPlaneResources(drmfd);
+		if (!planes) {
+			ERR("Cannot get drm planes: %d", errno);
```

Comment:
Is there a good reason not to print `errstring` with this? Similarly for other uses of errno.

Reply:
Not really. It's a useful suggestion, thanks!